src/controllers/collisions.rs

```diff
@@ -7,15 +11,26 @@ const SCORE_PER_ENEMY: u32 = 10;
 pub struct CollisionsController;
 
 impl CollisionsController {
-    pub fn handle_collisions(state: &mut GameState) {
-        CollisionsController::handle_bullet_collisions(state);
-        CollisionsController::handle_player_collisions(state);
+    pub fn handle_collisions(app: &mut ApplicationState, ctx: &mut Context) {
+        // Don't handle collisions if the player is dead
+        if app.game_state.world.player.is_dead { return; }
+
+        CollisionsController::handle_bullet_collisions(&mut app.game_state);
+        CollisionsController::handle_player_collisions(&mut app.game_state);
+        
+        // If the player died then we set a timeout (3 seconds) after which a game over message
+        // will appear, and the user will be able to restart.
+        if app.game_state.world.player.is_dead {
+            fn handler(app: &mut ApplicationState) { app.game_state.game_over(); };
+            let when = ggez::timer::get_time_since_start(ctx) + Duration::from_secs(3);
+            app.scheduled_events.push(when, handler);
```

Comment:
Note that closures that don't capture any variables are automatically promoted to functions. Therefore you can use `app.scheduled_events.push(when, |app| app.game_state.game_over());` here.

Reply:
Oh sweet thanks - didn't know that!
I feel like closures are the most complex things in Rust haha